indico/web/forms/widgets.py

```diff
@@ -198,6 +198,42 @@ def __call__(self, field, **kwargs):
                                                      search_url=self.search_url, choices=getattr(field, 'choices', []))
 
 
+class LocationWidget(JinjaWidget):
+    """Renders a collection of fields to represent location"""
+
+    def __init__(self):
+        super(LocationWidget, self).__init__('forms/location_widget.html', single_line=True)
+
+    def __call__(self, field, **kwargs):
+        if Config.getInstance().getIsRoomBookingActive():
+            rooms = {loc.name: {'data': self.get_sorted_rooms(loc)} for loc in field.locations}
+            venues = {'data': [{'id': loc.id, 'name': loc.name} for loc in field.locations]}
+        else:
+            rooms = venues = {'data': []}
+        parent = self._get_parent_info(field.data['source']) if field.data and field.data.get('source', None) else None
+        return super(LocationWidget, self).__call__(field, rooms=rooms, venues=venues, parent=parent)
+
+    def _get_parent_info(self, parent):
+        if isinstance(parent, db.m.Contribution):
+            return 'Contribution', parent.title
+        elif isinstance(parent, db.m.Break):
+            return 'Break', parent.title
+        elif isinstance(parent, db.m.SessionBlock):
+            return 'Block', parent.title
+        elif isinstance(parent, db.m.Session):
+            return 'Session', parent.title
+        elif isinstance(parent, db.m.Event):
+            return 'Event', parent.title
+        else:
+            raise TypeError('Unexpected parent type {}'.format(type(parent)))
+
+    def get_sorted_rooms(self, location):
+        result = []
+        for room in location.rooms:
+            result.append({'name': room.full_name, 'id': room.id, 'venue_id': room.location_id})
+        return sorted(result, key=lambda x: int(x['name'].split('-')[0]))
```

Comment:
Why not just `key=lambda x: natural_sort_key(x['name'])`? (we already have this function in indico)